src/Carbon/CarbonInterval.php

```diff
@@ -375,11 +375,11 @@ public static function make($var)
         if (is_string($var) && !empty($var)) {
             $var = trim($var);
 
-            if (preg_match('/^P/', $var)) {
+            if (substr($var, 0, 1) === 'P') {
```

Comment:
Even shorter code would be `$var[0] === 'P'` :)

Reply:
No thanks :) Reserve `[]` for arrays allow to identify array keys faster.